sympy/geometry/hyperbola.py

```diff
@@ -0,0 +1,986 @@
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Hyperbola
+        >>> p1 = Point(0, 0)
+        >>> h1 = Hyperbola(p1, 3, 1)
+        >>> h1.focus_distance
+        sqrt(10)
+
+        """
+        return Point.distance(self.center, self.foci[0])
+
+    @property
+    def hradius(self):
+        """The horizontal radius of the hyperbola.
+
+        Returns
+        =======
+
+        hradius : number
+
+        See Also
+        ========
+
+        vradius, major, minor
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Hyperbola
+        >>> p1 = Point(0, 0)
+        >>> h1 = Hyperbola(p1, 3, 1)
+        >>> h1.hradius
+        3
+
+        """
+        return self.args[1]
+
+    @property
+    def vradius(self):
+        """The vertical radius of the hyperbola.
+
+        Returns
+        =======
+
+        vradius : number
+
+        See Also
+        ========
+
+        hradius, major, minor
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Hyperbola
+        >>> p1 = Point(0, 0)
+        >>> h1 = Hyperbola(p1, 3, 1)
+        >>> h1.vradius
+        1
+
+        """
+        return self.args[2]
+
+    @property
+    def major(self):
+        """Longer axis of the hyperbola (if it can be determined) else hradius.
+
+        Returns
+        =======
+
+        major : number or expression
+
+        See Also
+        ========
+
+        hradius, vradius, minor
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Hyperbola, Symbol
+        >>> p1 = Point(0, 0)
+        >>> h1 = Hyperbola(p1, 3, 1)
+        >>> h1.major
+        3
+
+        >>> a = Symbol('a')
+        >>> b = Symbol('b')
+        >>> Hyperbola(p1, a, b).major
+        a
+        >>> Hyperbola(p1, b, a).major
+        b
+
+        >>> m = Symbol('m')
+        >>> M = m + 1
+        >>> Hyperbola(p1, m, M).major
+        m + 1
+
+        """
+        ab = self.args[1:3]
+        if len(ab) == 1:
+            return ab[0]
+        a, b = ab
+        o = b - a < 0
+        if o == True:
+            return a
+        elif o == False:
+            return b
+        return self.hradius
+
+    @property
+    def minor(self):
+        """Shorter axis of the hyperbola (if it can be determined) else vradius.
+
+        Returns
+        =======
+
+        minor : number or expression
+
+        See Also
+        ========
+
+        hradius, vradius, major
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Hyperbola, Symbol
+        >>> p1 = Point(0, 0)
+        >>> h1 = Hyperbola(p1, 3, 1)
+        >>> h1.minor
+        1
+
+        >>> a = Symbol('a')
+        >>> b = Symbol('b')
+        >>> Hyperbola(p1, a, b).minor
+        b
+        >>> Hyperbola(p1, b, a).minor
+        a
+
+        >>> m = Symbol('m')
+        >>> M = m + 1
+        >>> Hyperbola(p1, m, M).minor
+        m
+
+        """
+        ab = self.args[1:3]
+        if len(ab) == 1:
```

Comment:
I do not think this case is tested. (And I think the line above will give an error for this case.)

Reply:
Yeah currently that won't be required , so we could get rid of that but we would be using it whenever the rectangular hyperbola class is implemented !